x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/documents_logic.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import {
+  IndicesGetMappingIndexMappingRecord,
+  MappingProperty,
+  SearchResponseBody,
+  SearchHit,
+} from '@elastic/elasticsearch/lib/api/types';
+
+import { Status } from '../../../../../common/types/api';
+
+import { MappingsApiLogic } from '../../api/mapping/mapping_logic';
+import { SearchDocumentsApiLogic } from '../../api/search/search_documents_logic';
+
+import { IndexNameLogic } from './index_name_logic';
+
+interface DocumentsLogicActions {
+  apiReset: typeof SearchDocumentsApiLogic.actions.apiReset;
+  makeMappingRequest: typeof MappingsApiLogic.actions.makeRequest;
+  makeRequest: typeof SearchDocumentsApiLogic.actions.makeRequest;
+  setSearchQuery(query: string): { query: string };
+}
+
+interface DocumentsLogicValues {
+  data: typeof SearchDocumentsApiLogic.values.data;
+  indexName: typeof IndexNameLogic.values.indexName;
+  isLoading: boolean;
+  mappingData: IndicesGetMappingIndexMappingRecord;
+  mappingStatus: Status;
+  query: string;
+  results: SearchHit[];
+  simplifiedMapping: Record<string, MappingProperty>;
+  status: Status;
+}
+
+export const DocumentsLogic = kea<MakeLogicType<DocumentsLogicValues, DocumentsLogicActions>>({
```

Comment:
@sphilipse I think we need to add new error state helper here right ?

Reply:
that means not :D hehe, I was mentioning the error handler stuff you did a few daya ago

Reply:
Ah no, those error states are more for communicating errors between API and frontend. Unless you're handling new backend errors here, but I don't think so.